Kernel/Locking/Mutex.cpp

```diff
@@ -242,9 +242,6 @@ void Mutex::unblock_waiters(Mode previous_mode)
     VERIFY(m_mode == Mode::Unlocked);
 
     m_blocked_thread_lists.with([&](auto& lists) {
-        if (lists.exclusive.is_empty() && lists.shared.is_empty())
-            return;
```

Comment:
I'm pretty sure this is just supposed to be an optimisation to skip all of the rest of the logic. You can probably just add `&& lists.exclusive_big_lock.is_empty()`

Reply:
"Al of the rest of the logic" is a very simple set of checks, including _an additional_ check if the shared list is empty. If you follow the logic, we effectively perform the same checks very early on.

Reply:
I mean it's not the same, it includes multiple (possible) function calls and conditionals, but I agree that this is effectively a micro-optimisation, so either is fine.